sdk/rust-language-support/examples/image-processing/src/main.rs

```diff
@@ -0,0 +1,70 @@
+//! A simple linear-regression example
+//!
+//! ## Context
+//!
+//! A relatively-low powered device (think e.g. an M-class microcontroller, or similar)
+//! wants to offload some heavy, complex calculation to a third-party---here represented by a linear
+//! regression problem.  The owner of the device wishes to ensure that the computation was
+//! faithfully executed---that is, the results are trustworthy and are definitely the output of a
+//! linear regression algorithm---and also are kept secret from third parties wishing to see what is
+//! happening on the device.
+//!
+//! Inputs:                  1.
+//! Assumed form of inputs:  one Pinecode-encoded `Vec<(f64, f64)>` representing a dataset of (x, y)
+//!                          co-ordinates from which a linear relationship is to be extracted.
+//! Ensured form of outputs: A Pinecode-encoded `LinearRegression` struct (see below) consisting of
+//!                          a gradient and a Y-intercept, representing the best linear fit for the
+//!                          input data.
+//!
+//! ## Authors
+//!
+//! The Veracruz Development Team.
+//!
+//! ## Licensing and copyright notice
+//!
+//! See the `LICENSE.markdown` file in the Veracruz root directory for
+//! information on licensing and copyright.
+
+extern crate image;
+
+use image::{GenericImageView, imageops, ImageFormat, io::Reader};
+use std::process::exit;
+use wasi_types::ErrNo;
+
+/// Reads two input sources from the Veracruz host: the first input source is assumed to be a vector
+/// of `AdvertisementViewer` structs, whilst the second is assumed to be a vector of `Customer`
+/// structs.  Fails with [`return_code::ErrorCode::BadInput`] if the number of inputs provided is
+/// not equal to 2, or if the inputs cannot be deserialized from Bincode.
+/// TODO: read image directly (stream)
+
+fn process_image() -> Result<(), wasi_types::ErrNo> {
+    // Use the open function to load an image from a Path.
+    // `open` returns a `DynamicImage` on success.
+    // By default, jpeg-decode reads JPEG images in parallel threads (cf. rayon), which aren't supported in WASM. The workaround is to read PNG images instead
+    let mut img = image::open("/test.png").map_err(|_| {
+        println!("Failure opening /test.png");
+        ErrNo::Proto
+    })?;
+
+    // Transform the image
+    let subimg = imageops::crop(&mut img, 0, 0, 100, 100);
+    println!("new dimensions: {:?}", subimg.dimensions());
+
+    // Write the contents of this image to the Writer in PNG format.
+    subimg.to_image().save_with_format("/output", ImageFormat::Png).unwrap();
```

Comment:
Do we want an `unwrap`, and therefore a runtime panic, or some more graceful error notification, here?